### PR TITLE
[Bot-Ai-QnA] Add ActiveLearningUtils, BindToActivity and QnATelemetryConstant

### PR DESCRIPTION
**Pull request by Batta32**
Fixes #

## Specific Changes

  - 
  -
  -

### CODE REVIEW

File: libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/ActiveLearningUtils.java [hunk 1]
```diff
@@ -1,2 +1,104 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+
+package com.microsoft.bot.ai.qna.utils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.microsoft.bot.ai.qna.models.QueryResult;
```
Comment (fran893) [inline libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/ActiveLearningUtils.java:9]: Shouldn't this import go on top of the others?

File: libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/ActiveLearningUtils.java [hunk 2]
```diff
@@ -1,2 +1,104 @@
...
+
+    /**
+     * Returns list of qnaSearch results which have low score variation.
+     * @param qnaSearchResults List of QnaSearch results.
+     * @return List of filtered qnaSearch results.
+     */
+    public static List<QueryResult> getLowScoreVariation(List<QueryResult> qnaSearchResults) {
+        List<QueryResult> filteredQnaSearchResult = new ArrayList<QueryResult>();
+
+        if (qnaSearchResults.isEmpty()) {
```
Comment (fran893) [inline libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/ActiveLearningUtils.java:70]: Shouldn't control if the list is null? I think if the list is null this part will not work. See [here](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder.AI.QnA/Utils/ActiveLearningUtils.cs#L49)

File: libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/BindToActivity.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.bot.ai.qna.utils;
+
+import java.util.concurrent.CompletableFuture;
+
+import javax.annotation.Nullable;
+
+import com.microsoft.bot.schema.Activity;
```
Comment (fran893) [inline libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/BindToActivity.java:10]: This import should be on top

**Review by fran893 — CHANGES_REQUESTED** [review]

File: libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/BindToActivity.java (hunk 3 above)
Comment (Batta32) [reply libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/BindToActivity.java:10]: Done

File: libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/ActiveLearningUtils.java (hunk 1 above)
Comment (Batta32) [reply libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/ActiveLearningUtils.java:9]: Done

File: libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/ActiveLearningUtils.java (hunk 2 above)
Comment (Batta32) [reply libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/ActiveLearningUtils.java:70]: Agree with you

File: libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/QnACardBuilder.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.bot.ai.qna.utils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import com.google.common.base.Strings;
```
Comment (fran893) [inline libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/QnACardBuilder.java:10]: This import should go in the end after a blank line

File: libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/QnACardBuilder.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+     * @param cardTitle       Title of the cards
+     * @param cardNoMatchText No match text.
+     * @return Activity.
+     */
+    public static Activity getSuggestionsCard(List<String> suggestionsList, String cardTitle, String cardNoMatchText) {
+        if (suggestionsList == null) {
+            throw new IllegalArgumentException("suggestionsList");
+        }
+
+        if (Strings.isNullOrEmpty(cardTitle)) {
```
Comment (fran893) [inline libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/QnACardBuilder.java:35]: Here, in the [dotnet sample](https://github.com/microsoft/botbuilder-dotnet/blob/main/libraries/Microsoft.Bot.Builder.AI.QnA/Utils/QnACardBuilder.cs#L31), it verify only for null

File: libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/QnACardBuilder.java [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+     * @param result          Result to be dispalyed as prompts.
+     * @param cardNoMatchText No match text.
+     * @return Activity.
+     */
+    public static Activity getQnAPromptsCard(QueryResult result, String cardNoMatchText) {
+        if (result == null) {
+            throw new IllegalArgumentException("result");
+        }
+
+        if (Strings.isNullOrEmpty(cardNoMatchText)) {
```
Comment (fran893) [inline libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/QnACardBuilder.java:93]: Same as above

File: libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/QnACardBuilder.java [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+    public static Activity getSuggestionsCard(List<String> suggestionsList, String cardTitle, String cardNoMatchText) {
+        if (suggestionsList == null) {
+            throw new IllegalArgumentException("suggestionsList");
+        }
+
+        if (Strings.isNullOrEmpty(cardTitle)) {
+            throw new IllegalArgumentException("cardTitle");
+        }
+
+        if (Strings.isNullOrEmpty(cardNoMatchText)) {
```
Comment (fran893) [inline libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/QnACardBuilder.java:39]: Same as above

File: libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/TrainUtils.java [hunk 8]
```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.bot.ai.qna.utils;
+
+import java.util.concurrent.CompletableFuture;
+
+import com.microsoft.bot.ai.qna.models.FeedbackRecords;
```
Comment (fran893) [inline libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/TrainUtils.java:8]: These imports should be go at the beginning

**Review by fran893 — CHANGES_REQUESTED** [review]
A little feedback

File: libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/QnACardBuilder.java (hunk 4 above)
Comment (Batta32) [reply libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/QnACardBuilder.java:10]: Done

File: libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/QnACardBuilder.java (hunk 5 above)
Comment (Batta32) [reply libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/QnACardBuilder.java:35]: Done

File: libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/QnACardBuilder.java (hunk 6 above)
Comment (Batta32) [reply libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/QnACardBuilder.java:93]: Done

File: libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/QnACardBuilder.java (hunk 7 above)
Comment (Batta32) [reply libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/QnACardBuilder.java:39]: Done

File: libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/TrainUtils.java (hunk 8 above)
Comment (Batta32) [reply libraries/bot-ai-qna/src/main/java/com/microsoft/bot/ai/qna/utils/TrainUtils.java:8]: Done

**Review by VictorGrycuk — APPROVED** [review]

**Pull request merged**